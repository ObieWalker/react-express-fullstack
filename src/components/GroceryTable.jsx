import React from 'react';

const GroceryTable = ({groceries, deleteGrocery, buyGrocery, onShow}) => {
    return (
      <div>
        {groceries &&  groceries.length > 0 ? 
          <table className="table text-center table-hover mx-auto bg-white table-responsive-sm table-striped" style={{ marginLeft: '25%', marginRight: '10%', width: '50%' }}>
          <thead className="text-center text-white bg-info border border-white">
            <tr className="p-3">
              <th scope="col" className="border border-white">
                S/N
              </th>
              <th scope="col" className="border border-white">
                Grocery Name
              </th>
              <th scope="col" className="border border-white">
                Price
              </th>
              <th scope="col" className="border border-white"></th>
              <th scope="col" className="border border-white"></th>
              <th scope="col" className="border border-white"></th>
            </tr>
          </thead>
          <tbody>
          {groceries.map((grocery, i) =>
            <tr id="#1"
              key={i}
              index={i}
              className="border border-white text-left">
              <td className="border border-white">{i + 1}</td>
              <td>{grocery.name}</td>
              <td>&#8358; {grocery.price}</td>
              <td>
                <input
                type="submit"
                value={grocery.purchased ? 'Remove' : 'Purchase'}
                className={grocery.purchased ? 'btn btn-warning' : 'btn btn-primary'}
                onClick={() => buyGrocery(grocery._id, grocery.purchased)}
                />
              </td>
              <td>
                <button type="button" className="btn-warning btn-sm"
                onClick={() => onShow(grocery)}
                >
                  Edit Product
                </button>
              </td>
              <td>
                <button type="button" className="btn-danger btn-sm"
                onClick={() => deleteGrocery(grocery._id)}>
                  Delete Product
                </button>
              </td>
            </tr>
          )}
          </tbody>
        </table>
        : 
        <div>
        <h3 className="text-center ">No groceries Available at this time. Try reloading or add some.</h3>
        <br />
        </div>
        }
      </div>
    );
  }

export default GroceryTable;
