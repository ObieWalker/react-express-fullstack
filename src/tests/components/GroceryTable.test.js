import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import GroceryTable from '../../components/GroceryTable';

let mounted;
let props;
let wrapper;
const groceryName = ''
const groceryPrice = ''
const errors = {
  groceryName : '',
  groceryPrice: ''
}

const onShow = true;
const buyGrocery = jest.fn(() => Promise.resolve({}));
const deleteGrocery = jest.fn(() => Promise.resolve({}));
const  groceries = [{}]

const getComponent = () => {
  if (!mounted) {
    props = {
      groceries,
      deleteGrocery,
      buyGrocery,
      onShow
    };
    mounted = shallow(<GroceryTable {...props} />);
  }
  return mounted;
};

describe('GroceryModal Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

});

