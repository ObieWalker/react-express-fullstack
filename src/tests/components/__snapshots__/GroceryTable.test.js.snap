// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroceryModal Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroceryTable
    buyGrocery={[MockFunction]}
    deleteGrocery={[MockFunction]}
    groceries={
      Array [
        Object {},
      ]
    }
    onShow={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="table text-center table-hover mx-auto bg-white table-responsive-sm table-striped"
        style={
          Object {
            "marginLeft": "25%",
            "marginRight": "10%",
            "width": "50%",
          }
        }
      >
        <thead
          className="text-center text-white bg-info border border-white"
        >
          <tr
            className="p-3"
          >
            <th
              className="border border-white"
              scope="col"
            >
              S/N
            </th>
            <th
              className="border border-white"
              scope="col"
            >
              Grocery Name
            </th>
            <th
              className="border border-white"
              scope="col"
            >
              Price
            </th>
            <th
              className="border border-white"
              scope="col"
            />
            <th
              className="border border-white"
              scope="col"
            />
            <th
              className="border border-white"
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr
            className="border border-white text-left"
            id="#1"
            index={0}
          >
            <td
              className="border border-white"
            >
              1
            </td>
            <td />
            <td>
              ₦ 
            </td>
            <td>
              <input
                className="btn btn-primary"
                onClick={[Function]}
                type="submit"
                value="Buy"
              />
            </td>
            <td>
              <button
                className="btn-warning btn-sm"
                onClick={[Function]}
                type="button"
              >
                Edit Product
              </button>
            </td>
            <td>
              <button
                className="btn-danger btn-sm"
                onClick={[Function]}
                type="button"
              >
                Delete Product
              </button>
            </td>
          </tr>
        </tbody>
      </table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead
            className="text-center text-white bg-info border border-white"
          >
            <tr
              className="p-3"
            >
              <th
                className="border border-white"
                scope="col"
              >
                S/N
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Grocery Name
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Price
              </th>
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
            </tr>
          </thead>,
          <tbody>
            <tr
              className="border border-white text-left"
              id="#1"
              index={0}
            >
              <td
                className="border border-white"
              >
                1
              </td>
              <td />
              <td>
                ₦ 
              </td>
              <td>
                <input
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="submit"
                  value="Buy"
                />
              </td>
              <td>
                <button
                  className="btn-warning btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  Edit Product
                </button>
              </td>
              <td>
                <button
                  className="btn-danger btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  Delete Product
                </button>
              </td>
            </tr>
          </tbody>,
        ],
        "className": "table text-center table-hover mx-auto bg-white table-responsive-sm table-striped",
        "style": Object {
          "marginLeft": "25%",
          "marginRight": "10%",
          "width": "50%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr
              className="p-3"
            >
              <th
                className="border border-white"
                scope="col"
              >
                S/N
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Grocery Name
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Price
              </th>
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
            </tr>,
            "className": "text-center text-white bg-info border border-white",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  className="border border-white"
                  scope="col"
                >
                  S/N
                </th>,
                <th
                  className="border border-white"
                  scope="col"
                >
                  Grocery Name
                </th>,
                <th
                  className="border border-white"
                  scope="col"
                >
                  Price
                </th>,
                <th
                  className="border border-white"
                  scope="col"
                />,
                <th
                  className="border border-white"
                  scope="col"
                />,
                <th
                  className="border border-white"
                  scope="col"
                />,
              ],
              "className": "p-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "S/N",
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "S/N",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Grocery Name",
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Grocery Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Price",
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Price",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": null,
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": null,
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "border border-white",
                  "scope": "col",
                },
                "ref": null,
                "rendered": null,
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr
                className="border border-white text-left"
                id="#1"
                index={0}
              >
                <td
                  className="border border-white"
                >
                  1
                </td>
                <td />
                <td>
                  ₦ 
                </td>
                <td>
                  <input
                    className="btn btn-primary"
                    onClick={[Function]}
                    type="submit"
                    value="Buy"
                  />
                </td>
                <td>
                  <button
                    className="btn-warning btn-sm"
                    onClick={[Function]}
                    type="button"
                  >
                    Edit Product
                  </button>
                </td>
                <td>
                  <button
                    className="btn-danger btn-sm"
                    onClick={[Function]}
                    type="button"
                  >
                    Delete Product
                  </button>
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td
                    className="border border-white"
                  >
                    1
                  </td>,
                  <td />,
                  <td>
                    ₦ 
                  </td>,
                  <td>
                    <input
                      className="btn btn-primary"
                      onClick={[Function]}
                      type="submit"
                      value="Buy"
                    />
                  </td>,
                  <td>
                    <button
                      className="btn-warning btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Edit Product
                    </button>
                  </td>,
                  <td>
                    <button
                      className="btn-danger btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete Product
                    </button>
                  </td>,
                ],
                "className": "border border-white text-left",
                "id": "#1",
                "index": 0,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                    "className": "border border-white",
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "₦ ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "₦ ",
                    undefined,
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="btn btn-primary"
                      onClick={[Function]}
                      type="submit"
                      value="Buy"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "btn btn-primary",
                      "disabled": undefined,
                      "onClick": [Function],
                      "type": "submit",
                      "value": "Buy",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-warning btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Edit Product
                    </button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edit Product",
                      "className": "btn-warning btn-sm",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Edit Product",
                    "type": "button",
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-danger btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete Product
                    </button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Delete Product",
                      "className": "btn-danger btn-sm",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Delete Product",
                    "type": "button",
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="table text-center table-hover mx-auto bg-white table-responsive-sm table-striped"
          style={
            Object {
              "marginLeft": "25%",
              "marginRight": "10%",
              "width": "50%",
            }
          }
        >
          <thead
            className="text-center text-white bg-info border border-white"
          >
            <tr
              className="p-3"
            >
              <th
                className="border border-white"
                scope="col"
              >
                S/N
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Grocery Name
              </th>
              <th
                className="border border-white"
                scope="col"
              >
                Price
              </th>
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
              <th
                className="border border-white"
                scope="col"
              />
            </tr>
          </thead>
          <tbody>
            <tr
              className="border border-white text-left"
              id="#1"
              index={0}
            >
              <td
                className="border border-white"
              >
                1
              </td>
              <td />
              <td>
                ₦ 
              </td>
              <td>
                <input
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="submit"
                  value="Buy"
                />
              </td>
              <td>
                <button
                  className="btn-warning btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  Edit Product
                </button>
              </td>
              <td>
                <button
                  className="btn-danger btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  Delete Product
                </button>
              </td>
            </tr>
          </tbody>
        </table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead
              className="text-center text-white bg-info border border-white"
            >
              <tr
                className="p-3"
              >
                <th
                  className="border border-white"
                  scope="col"
                >
                  S/N
                </th>
                <th
                  className="border border-white"
                  scope="col"
                >
                  Grocery Name
                </th>
                <th
                  className="border border-white"
                  scope="col"
                >
                  Price
                </th>
                <th
                  className="border border-white"
                  scope="col"
                />
                <th
                  className="border border-white"
                  scope="col"
                />
                <th
                  className="border border-white"
                  scope="col"
                />
              </tr>
            </thead>,
            <tbody>
              <tr
                className="border border-white text-left"
                id="#1"
                index={0}
              >
                <td
                  className="border border-white"
                >
                  1
                </td>
                <td />
                <td>
                  ₦ 
                </td>
                <td>
                  <input
                    className="btn btn-primary"
                    onClick={[Function]}
                    type="submit"
                    value="Buy"
                  />
                </td>
                <td>
                  <button
                    className="btn-warning btn-sm"
                    onClick={[Function]}
                    type="button"
                  >
                    Edit Product
                  </button>
                </td>
                <td>
                  <button
                    className="btn-danger btn-sm"
                    onClick={[Function]}
                    type="button"
                  >
                    Delete Product
                  </button>
                </td>
              </tr>
            </tbody>,
          ],
          "className": "table text-center table-hover mx-auto bg-white table-responsive-sm table-striped",
          "style": Object {
            "marginLeft": "25%",
            "marginRight": "10%",
            "width": "50%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr
                className="p-3"
              >
                <th
                  className="border border-white"
                  scope="col"
                >
                  S/N
                </th>
                <th
                  className="border border-white"
                  scope="col"
                >
                  Grocery Name
                </th>
                <th
                  className="border border-white"
                  scope="col"
                >
                  Price
                </th>
                <th
                  className="border border-white"
                  scope="col"
                />
                <th
                  className="border border-white"
                  scope="col"
                />
                <th
                  className="border border-white"
                  scope="col"
                />
              </tr>,
              "className": "text-center text-white bg-info border border-white",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    className="border border-white"
                    scope="col"
                  >
                    S/N
                  </th>,
                  <th
                    className="border border-white"
                    scope="col"
                  >
                    Grocery Name
                  </th>,
                  <th
                    className="border border-white"
                    scope="col"
                  >
                    Price
                  </th>,
                  <th
                    className="border border-white"
                    scope="col"
                  />,
                  <th
                    className="border border-white"
                    scope="col"
                  />,
                  <th
                    className="border border-white"
                    scope="col"
                  />,
                ],
                "className": "p-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "S/N",
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "S/N",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Grocery Name",
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Grocery Name",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Price",
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Price",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "border border-white",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr
                  className="border border-white text-left"
                  id="#1"
                  index={0}
                >
                  <td
                    className="border border-white"
                  >
                    1
                  </td>
                  <td />
                  <td>
                    ₦ 
                  </td>
                  <td>
                    <input
                      className="btn btn-primary"
                      onClick={[Function]}
                      type="submit"
                      value="Buy"
                    />
                  </td>
                  <td>
                    <button
                      className="btn-warning btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Edit Product
                    </button>
                  </td>
                  <td>
                    <button
                      className="btn-danger btn-sm"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete Product
                    </button>
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td
                      className="border border-white"
                    >
                      1
                    </td>,
                    <td />,
                    <td>
                      ₦ 
                    </td>,
                    <td>
                      <input
                        className="btn btn-primary"
                        onClick={[Function]}
                        type="submit"
                        value="Buy"
                      />
                    </td>,
                    <td>
                      <button
                        className="btn-warning btn-sm"
                        onClick={[Function]}
                        type="button"
                      >
                        Edit Product
                      </button>
                    </td>,
                    <td>
                      <button
                        className="btn-danger btn-sm"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete Product
                      </button>
                    </td>,
                  ],
                  "className": "border border-white text-left",
                  "id": "#1",
                  "index": 0,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 1,
                      "className": "border border-white",
                    },
                    "ref": null,
                    "rendered": 1,
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "₦ ",
                        undefined,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "₦ ",
                      undefined,
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        className="btn btn-primary"
                        onClick={[Function]}
                        type="submit"
                        value="Buy"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "btn btn-primary",
                        "disabled": undefined,
                        "onClick": [Function],
                        "type": "submit",
                        "value": "Buy",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn-warning btn-sm"
                        onClick={[Function]}
                        type="button"
                      >
                        Edit Product
                      </button>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Edit Product",
                        "className": "btn-warning btn-sm",
                        "onClick": [Function],
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": "Edit Product",
                      "type": "button",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn-danger btn-sm"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete Product
                      </button>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Delete Product",
                        "className": "btn-danger btn-sm",
                        "onClick": [Function],
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": "Delete Product",
                      "type": "button",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
